### ISSUE TITLE
potential improvements to palThreadPriority_t

### ISSUE THREAD

**Issue by rgrover** [ARMmbed/pal#34]
``` C++
//! Available priorities in PAL implementation, each priority can appear only once.
typedef enum    pal_osPriority {
    PAL_osPriorityIdle = -3,
    PAL_osPriorityLow = -2,
    PAL_osPriorityBelowNormal = -1,
    PAL_osPriorityNormal = 0,
    PAL_osPriorityAboveNormal = +1,
    PAL_osPriorityHigh = +2,
    PAL_osPriorityRealtime = +3,
    PAL_osPriorityError = 0x84
} palThreadPriority_t; /*! Thread priority levels for PAL threads - each thread must have a different priority*/
```
- The name `PAL_osPriorityError` suggests that this is an error, whereas it is actually an _INVALID priority_. Perhaps `PAL_osPriorityInvalid` would be a better name.
- If every thread needs to have a distinct priority, then this enumeration might allow for very few threads to exist. This could be a severe limitations for some applications.
- Why is `PAL_osPriorityError` selected to be `0x84`? Why is it positive? And what is special about this value? If there is a deliberate intention to limit priorities to the range [-3...3], having a smaller value for `PAL_osPriorityError` might allow encoding `palThreadPriority_t` in 5 bits.

**Comment by rgrover** [comment]
In the implementation of the MBED-OS port, there are snippets like `priority > PAL_osPriorityRealtime`. Code like this assumes that real-time is the highest priority. Are these assumptions future-proof?

**Comment by rgrover** [comment]
Shouldn't the available priorities be platform specific? Why is the PAL attempting to impose a definition? Why don't we simply introduce helper functions for validity and comparison of priorities? `bool comparePriority(palThreadPriority_t, palThreadPriority_t)` and `isValidPriority(palThreadPriority_t)`.

**Comment by teetak01** [comment]
Closing the issue as inactive and due to the repository being deprecated.

Please take a look at our current offering at https://cloud.mbed.com/docs/current. The actively maintained version of this component can be found in http://github.com/armmbed/mbed-cloud-client.

Please open a new issue if this is still valid. We strongly encourage you to migrate to the latest product. New issues in this repository will most likely remain unfixed.
